se/src/main/java/org/jboss/forge/furnace/se/BootstrapClassLoader.java

```diff
@@ -110,7 +110,11 @@ private List<URL> handleZipStream(URL original) throws IOException, FileNotFound
                   try
                   {
                      byte[] buffer = new byte[2048];
-                     output = new FileOutputStream(new File(tempDir, entry.getName()));
+                     final File zipEntryFile = new File(tempDir, entry.getName());
+                     if (!zipEntryFile.toPath().normalize().startsWith(tempDir.toPath().normalize())) {
```

Comment:
Use the Forge code format: https://github.com/forge/core/blob/master/eclipse-code-formatter-profile.xml